src/asmcnc/apps/wifi_app/screen_wifi.py

```diff
@@ -476,6 +476,23 @@ def connect_wifi(self):
             os.system('echo "update_config=1" | sudo tee --append /etc/wpa_supplicant/wpa_supplicant-wlan0.conf')
             os.system('echo "country="' + self.country.text + '| sudo tee --append /etc/wpa_supplicant/wpa_supplicant-wlan0.conf')
 
+            # Flush all the IP addresses from cache
+            os.system('sudo ip addr flush dev wlan0')
+
+            # Reload the updated wpa_supplicant file
+            os.system('sudo wpa_cli -i wlan0 reconfigure')
+
+            # Restart the DHCP service to allocate a new IP address on the new network
+            os.system('sudo systemctl restart dhcpcd')
+
+            def dismiss_wait_popup(dt):
+                if self.set.wifi_available:
+                    wait_popup.popup.dismiss()
+                    return
+                Clock.schedule_once(dismiss_wait_popup, 0.5)

```

Comment:
This does need a timeout/fail condition (users will type in details wrong sometimes or possibly intentionally type in wrong creds to disconnect) 

"No WiFi connection!" has already been translated, so I recommend using that as the text on the error popup for now. 

It's worth adding some extra text for this error in the near future as well (along the lines of check your network name/password/try again/check your network/contact yetitool etc.), so I'd recommend writing the full error message and getting it translated asap. However, "No WiFi connection!" will pass for now, so that can be treated as a separate ticket on a new branch :)

Reply:
- I've now changed it so that the new block of code gets called after all try-except blocks.
- Due to the timeout, unless the connection is successful, the error popup will come up and warn the user that something went wrong.